pkg/clusterapi/aws/actuator.go

```diff
@@ -425,41 +416,20 @@ func (a *Actuator) clusterOperatorClusterSpec(c *clusterv1.Cluster) (*cov1.Clust
 	return coProviderConfig, nil
 }
 
-func (a *Actuator) clusterOperatorMachineSet(m *clusterv1.Machine) (*cov1.MachineSet, *cov1.ClusterVersion, error) {
-	if m.Spec.ProviderConfig.Value == nil || m.Spec.ProviderConfig.ValueFrom == nil {
-		return nil, nil, fmt.Errorf("No providerConfig specified for machine")
-	}
-	obj, _, err := a.codecFactory.UniversalDecoder(cov1.SchemeGroupVersion).Decode(m.Spec.ProviderConfig.Value.Raw, nil, nil)
-	if err != nil {
-		return nil, nil, err
-	}
-	coMachineSet, ok := obj.(*cov1.MachineSet)
-	if !ok {
-		return nil, nil, fmt.Errorf("Unexpected machine set object: %#v", obj)
-	}
-	rawClusterVersion, ok := coMachineSet.Annotations[clusterVersionAnnotation]
-	if !ok {
-		return nil, nil, fmt.Errorf("Missing ClusterVersion resource annotation in MachineSet %#v", coMachineSet)
+func (a *Actuator) clusterOperatorMachineSetSpec(m *clusterv1.Machine) (*cov1.MachineSetSpec, error) {
```

Comment:
Can we use `MachineSetSpecFromClusterAPIMachineSpec` from `pkg/controller/controller_utils.go` instead of defining this function?

Reply:
Almost surely, this is quite old and was written in a separate repo IIRC.